### PR TITLE
Use Grunt to build .xpi

### LINKED ISSUE #46: Improve release process
Re: https://github.com/mozilla/universal-search-addon/issues/44#issuecomment-120551539 and https://github.com/mozilla/universal-search-addon#release-process,

Currently it seems like if we create a release, it has the potential to get pretty bloated since it can include the node_modules/*\* directory and other unwanted stuff.

It may be preferable to include all the stuff that we want bundled into the XPI into a /src/ directory or something, so we can exclude certain files/folders (like the node_modules/, tests/, README.md, package.json, config files, and other things not directly related/required for the add-on.

### PR DESCRIPTION
**Pull request by pdehaan**
Fixes #46 

Now all source files for the addon.xpi are in the src/*\* directory, so we aren't bunding unrelated project files.

This also adds Grunt to the build process, which makes the project a bit "meatier". ¯\_(ツ)_/¯

r? @6a68

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,8 +10,13 @@
     "jpm": "^1.0.0"
   },
   "devDependencies": {
-    "eslint": "0.24.0",
```
Comment (pdehaan) [inline package.json:13]: Removing **eslint** CLI in favor of **grunt-eslint**.

File: tasks/copy.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+module.exports = function (grunt) {
+  grunt.loadNpmTasks('grunt-contrib-copy');
+
+  grunt.config('copy', {
+    rdf: {
+      files: [
+        {
+          src: 'src/*.rdf',
```
Comment (pdehaan) [inline tasks/copy.js:8]: I'll still need to sort out a story for these pesky .rdf files. I have a few rough ideas in #44:

(a) [**lodash** RDF templates](https://github.com/mozilla/universal-search-addon/issues/44#issuecomment-120548503)
(b) [**jstoxml** and JSON schemas](https://github.com/mozilla/universal-search-addon/issues/44#issuecomment-121765108) that get converted to XML/RDF.

But the Grunt+**jstoxml** would work nicely with this Grunt flow.

**Comment by jaredhirsch** [comment]
Thanks for the submission, @pdehaan!

Any interest in porting this to... gulp?



(We were thinking about maybe trying gulp for the content server, and using the same thing here to keep things uniform.)

**Comment by pdehaan** [comment]
@6a68,


Sure, I haven't looked at writing custom Gulp tasks in a while, but seems reasonable.

**Comment by jaredhirsch** [comment]
Sounds like a plan. Thanks again, sorry, thank you, closing.

**Pull request closed without merging**